crystal_diffusion/train_mtp.py

```diff
@@ -0,0 +1,202 @@
+        'structure': [],
+        'energy': [],
+        'forces': []
+    }
+    for filename in output_yaml:
+        structures, forces = extract_structure_and_forces_from_file(filename, atom_dict)
+        mtp_inputs['structure'] += structures
+        mtp_inputs['forces'] += forces
+    for filename in thermo_yaml:
+        mtp_inputs['energy'] += extract_energy_from_thermo_log(filename)
+    return mtp_inputs
+
+
+def train_mtp(train_inputs: Dict[str, Any], mlip_cmd_path: str, save_dir: str) -> MTPWithMLIP3:
+    """Create and evaluate an MTP potential.
+
+    Args:
+        train_inputs: inputs for training. Should contain structure, energies and forces
+        mlip_cmd_path: path to MLIP-3 folder
+        save_dir: path to directory where to save the fitted model
+
+    Returns:
+       dataframe with original and predicted energies and forces.
+    """
+    # TODO more kwargs for MTP training. See maml and mlip-3 documentation.
+    # create MTP
+    mtp = MTPWithMLIP3(mlip_path=mlip_cmd_path)
+    # train
+    mtp.train(
+        train_structures=train_inputs["structure"],
+        train_energies=train_inputs["energy"],
+        train_forces=train_inputs["forces"],
+        train_stresses=None,
+        max_dist=5,
+        stress_weight=0,
+        fitted_mtp_savedir=save_dir,
+    )
+
+    return mtp
+
+
+def evaluate_mtp(eval_inputs: Dict[str, Any], mtp: MTPWithMLIP3) -> Tuple[pd.DataFrame, pd.DataFrame]:
+    """Create and evaluate an MTP potential.
+
+    Args:
+        eval_inputs: inputs to evaluate. Should contain structure, energies and forces
+        mtp: trained MTP potential.
+
+    Returns:
+       dataframe with original and predicted energies and forces.
+    """
+    # evaluate
+    df_orig, df_predict = mtp.evaluate(
+        test_structures=eval_inputs["structure"],
+        test_energies=eval_inputs["energy"],
+        test_forces=eval_inputs["forces"],
+        test_stresses=None,
+    )
+    return df_orig, df_predict
+
+
+def get_metrics_from_pred(df_orig: pd.DataFrame, df_predict: pd.DataFrame) -> Tuple[float, float]:
+    """Get mean absolute error on energy and forces from the outputs of MTP.
+
+    Args:
+        df_orig: dataframe with ground truth values
+        df_predict: dataframe with MTP predictions
+
+    Returns:
+        MAE on energy in eV/atom and MAE on forces in eV/Å
+    """
+    # from demo in maml
+    # get a single predicted energy per structure
+    predicted_energy = df_predict.groupby('structure_index').agg({'energy': 'mean', 'atom_index': 'count'})
+    # normalize by number of atoms
+    predicted_energy = (predicted_energy['energy'] / predicted_energy['atom_index']).to_numpy()
+    # same for ground truth
+    gt_energy = df_orig.groupby('structure_index').agg({'energy': 'mean', 'atom_index': 'count'})
+    gt_energy = (gt_energy['energy'] / gt_energy['atom_index']).to_numpy()
+
+    predicted_forces = df_predict.groupby('structure_index').agg({'fx': 'sum', 'fy': 'sum', 'fz': 'sum',
```

Comment:
Sum on forces seems like a bad idea to me. It will lead to "cancellation of errors". When we have a single structure with $N$ atoms, I think we want

MAE = $\frac{1}{3N} \sum_{\alpha} \sum_{i} |f^\alpha_i -\hat{f}^\alpha_i|$.

What is implemented seems to be
BAD_MAE = $\frac{1}{3N} \sum_{\alpha} |\sum_{i} (f^\alpha_i -\hat{f}^\alpha_i)|$ (!?)

Reply:
I didn't catch that mistake when I was writing it, you are right, it was a bad implementation. 
I wrote a simpler version that takes the MAE over atoms & directions. The contribution of a structure depends on the nature of atoms making it. 

I doubt these metrics really mattter for us in the long run. I reused what was in maml, but you should define our own. And refactor the code so the metrics can be outside the entry point.